resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -115,6 +116,19 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
     }
   }
 
+  if (isClusterMode) {
+    val nmHostPort = WebAppUtils.getWebAppBindURL(yarnConf, YarnConfiguration.NM_BIND_HOST,
+      WebAppUtils.getNMWebAppURLWithoutScheme(yarnConf))
+    val (nmHost, _) = Utils.parseHostPort(nmHostPort)
+
+    Seq(DRIVER_BIND_ADDRESS, DRIVER_HOST_ADDRESS)
```

Comment:
`DRIVER_BIND_ADDRESS` is something that is not set by Spark anywhere, so it feels a little weird to override it. If the user is setting that explicitly, the code should assume the user knows what he's doing.